projects/klippa/ngx-enhancy-forms/src/lib/form/form-validation-error/form-validation-error.ts

```diff
@@ -0,0 +1,36 @@
+import { InjectionToken } from '@angular/core';
+
+export class FormValidationError extends Error {
+	_path: string;
+	_message: string;
+
+	constructor(path: string, message: string) {
+		super(message);
+		this.name = 'FORM_VALIDATION_ERROR';
+		this._path = path;
+		this._message = message;
+	}
+
+	get path() {
+		return this._path;
+	}
+}
+
+export type FormValidationErrors = Array<FormValidationError>;
+
+export type FormErrorHandler = (error: any) => FormValidationErrors;
+
+export const KLP_FORM_ERROR_HANDLER = new InjectionToken<FormErrorHandler>('KLP_FORM_ERROR_HANDLER');
+
+export const DefaultErrorHandler: FormErrorHandler = (error: any) => {
+	if (Array.isArray(error) && error.reduce((acc, err) => acc && err instanceof FormValidationError)) {
```

Comment:
A reduce inside the condition is kinda hard to read. Also, this reducer does not have an initial value?

Reply:
I've added a utility for array typing.